File: backend_net/ChronosDtnNetApi/Exceptions/ResourceNotFoundException.cs
// Por que: Importa o namespace base do sistema.
using System;

// Por que: Organiza as exceções no namespace adequado do projeto.
namespace ChronosDtnNetApi.Exceptions;

// Por que: Exceção de negócio disparada sempre que um recurso (nó, conta, bundle ou transação) não for localizado.
public class ResourceNotFoundException : Exception
{
    // Por que: Construtor repassando o identificador do recurso ausente para a classe base Exception.
    public ResourceNotFoundException(string message) : base(message)
    {
        // Por que: Delega a mensagem explicativa para a classe mãe.
    }
}

File: backend_java/src/main/java/br/com/fiap/chronosdtn/exception/ResourceNotFoundException.java
package br.com.fiap.chronosdtn.exception;

// Por que: Permite lançar erros específicos quando recursos do gateway não são encontrados.
public class ResourceNotFoundException extends RuntimeException {
    
    // Por que: Construtor que recebe a mensagem descritiva do recurso ausente.
    public ResourceNotFoundException(String message) {
        // Por que: Passa a mensagem de erro para a classe base RuntimeException.
        super(message);
    }
}
